decompose/src/commonMain/kotlin/com/arkivanov/decompose/router/stack/webhistory/WebHistoryController.kt

```diff
@@ -36,5 +36,6 @@ interface WebHistoryController {
         serializer: KSerializer<C>,
         getPath: (configuration: C) -> String,
         getConfiguration: (path: String) -> C,
+        allowWebNavigationCallback: ((C, (Boolean)->Unit)->Unit)? = null
```

Comment:
It appears that this callback type could be just `(configuration: C) -> Boolean`, with the default value `= { true}`. In this case the name could be `onBrowserNavigation`. WDYT?

Another question: does it make sense to return the entire configuration stack as `List<C>` instead of just the new configuration?

Also, could you please document the new argument?

Reply:
The reason to use a second callback function (Boolean)->Unit is so that the allowWebNavigationCallback function can modify the UI state to ask for user input, e.g. to show a confirmation dialog box and then invoke the function with the result indicating if the navigation should be allowed or not.

Providing List<C> could be useful if the decision logic needs to take into consideration the complete back stack.

Reply:
I think we could still do that with `(configuration: C) -> Boolean`? Just update the UI state from the callback and return `true/false` from the callback.

Reply:
Maybe I am doing something incorrectly, or in an unusual way which is causing a complication? When I update the UI state from within the callback, the callback continues to execute to completion, but the response from the user isn't known yet. So what I am doing is updating the UI state, which on recomposition causes the dialog to be displayed. Then when the user presses the ok or cancel button in the dialog, it calls the (Boolean)->Unit function with true or false depending on what the user selected.

Reply:
Thanks for the explanation of the use case! Calling the callback eventually looks dangerous because the state may be changed (i.e. become invalid) at that point. Also the callback may be called multiple times by accident. 

What if we define the callback as `(configuration: C) -> Boolean`, and in that callback we make the decision whether we show the dialog or not. If we don't need the dialog, we return `true` and the browser navigation continues as usual. If we do need the dialog, we show the dialog and return `false`, so that the browser navigation is reverted. Then if the dialog is confirmed, we just navigation as usual via `Child Stack`. WDYT?

Reply:
Assuming you can derive this information in the callback, you could save this information somewhere if needed. The point is that saving might not be needed at all. But if you want, you can save the info in the dialog configuration, or in a variable, etc.

Reply:
Wouldn't it be better to do that derivation, which is currently done in ```doOnPopState()```, by the library instead of requiring the library user to implement it? We can also add safety by checking that the state when called back is the same as when onPopState was first invoked, and if it isn't the same don't perform the navigation. This result can be signaled back to the caller by changing the function signature from (Boolean)->Unit to (Boolean)->Boolean.

Reply:
My concern is that the browser history state stays out of sync with the Child Stack state while the dialog is being shown. I feels that ideally we should revert the history change if the stack navigation has not been performed accordingly. And then navigate as usual if the user confirms.

Another concern is that the dialog button click event comes separately (e.g. via a component's method like `onCloseDialogConfirmClicked`). And we'd have to store the callback somewhere for the time being.

Does it make sense?

Reply:
I see, I think I understand what you mean. We shouldn't keep the browser history and the Child Stack out of sync while the dialog is displayed and waiting for user input.

For the callback to compare between the current Child Stack and the browser stack, the callback will need to be passed the browser stack. Then the user code can compare it with the Child Stack like is done in the current onPopState function. Is that how you are thinking it would work in the callback?

Reply:
> For the callback to compare between the current Child Stack and the browser stack

I'm not 100% sure, what you mean. Could you please elaborate? It might be useful to pass `List<C>` to the callback, so that the component could decide what happened and how it should proceed.

Reply:
I just committed a change to implement an onBrowserNavigation(List<C>). Please check it and see if that is what you were suggesting.